python/ray/train/_internal/dl_predictor.py

```diff
@@ -72,14 +73,29 @@ def _predict_pandas(
         model_input = self._arrays_to_tensors(tensors, dtype)
 
         output = self.call_model(model_input)
-
-        # Handle model multi-output. For example if model outputs 2 images.
-        if isinstance(output, dict):
-            return pd.DataFrame(
-                {k: TensorArray(self._tensor_to_array(v)) for k, v in output.items()}
-            )
-        else:
-            return pd.DataFrame(
-                {"predictions": TensorArray(self._tensor_to_array(output))},
-                columns=["predictions"],
-            )
+        try:
+            # For predictor outputs we provide a default fall through behavior if
+            # user provides a pd.DataFrame as return type.
+            if isinstance(output, pd.DataFrame):
+                return output
+            # Handle model multi-output. For example if model outputs 2 images.
+            elif isinstance(output, dict):
+                return pd.DataFrame(
+                    {
+                        k: TensorArray(self._tensor_to_array(v))
+                        for k, v in output.items()
+                    }
+                )
+            else:
+                return pd.DataFrame(
+                    {"predictions": TensorArray(self._tensor_to_array(output))},
+                    columns=["predictions"],
+                )
+        except ValueError as e:
+            raise ValueError(
+                "Tried to cast output columns to the TensorArray tensor "
+                "extension type but the conversion failed. For your custom "
+                "Pytorch or TensorFlow predictor, consider structuring the output "
```

Comment:
Actually, I don't think it's appropriate to catch the ValueError here. We don't know if it's always due to TensorArray, it could be something else.

Can you remove this catch? We can instead improve the error message in the TensorArray constructor, or otherwise more closely match a narrow exception type (e.g., RaggedArrayNotSupported).

Reply:
makes sense, let me update it accordingly